iam/cloud-client/snippets/conftest.py

```diff
@@ -54,3 +56,36 @@ def delete_existing_deny_policies(project_id: str, delete_name_prefix: str) -> N
     for policy in policies_client.list_policies(request=request):
         if delete_name_prefix in policy.name:
             delete_deny_policy(PROJECT_ID, str(policy.name).rsplit("/", 1)[-1])
+
+@pytest.fixture
+def iam_role(capsys: "pytest.CaptureFixture[str]") -> str:
+    role_prefix = "test_iam_role"
+    role_id = f"{role_prefix}_{uuid.uuid4()}".replace("-", "_")
+    permissions = ["iam.roles.get", "iam.roles.list"]
+    title = "test_role_title"
+    # Delete any iam roles. Otherwise, it might throw quota issue.
+    delete_existing_iam_roles(PROJECT_ID, role_prefix)
+
+    # Create the iam role.
+    create_role(PROJECT_ID, role_id, permissions, title)
+    yield role_id
+
+    # Delete the iam role and assert if deleted.
+    delete_role(PROJECT_ID, role_id)
+    out, _ = capsys.readouterr()
+    assert re.search(f"Deleted role: {role_id}", out)
```

Comment:
Please use the following structure:

```python
try:
    create_role(...)
    created = True
    yield role_id
finally:
    if created:
        delete_role(...)
```
That way we make sure we at least attempt to clean up when something goes wrong in the test.